frame/benchmarking/src/analysis.rs

```diff
@@ -72,6 +108,46 @@ impl TryFrom<Option<String>> for AnalysisChoice {
 	}
 }
 
+fn raw_linear_regression(
+	xs: Vec<f64>,
+	ys: Vec<f64>,
+	x_vars: usize,
+	with_intercept: bool,
+) -> Option<(f64, Vec<f64>, Vec<f64>)> {
+	let mut data: Vec<f64> = Vec::new();
+	for (&y, xs) in ys.iter().zip(xs.chunks_exact(x_vars)) {
+		data.push(y);
+		if with_intercept {
+			data.push(1.0);
+		} else {
+			data.push(0.0);
+		}
```

Comment:
might be obvious, but dunno what this is doing atm

Reply:
I've added an extra comment; hopefully it's a little more clear.